SentinelOne/sentinelone_module/logs/connector.py

```diff
@@ -0,0 +1,123 @@
+from management.mgmtsdk_v2.exceptions import UnauthorizedException
+from management.mgmtsdk_v2.services.activity import ActivitiesFilter
+from management.mgmtsdk_v2.services.threat import ThreatQueryFilter
+from management.mgmtsdk_v2_1.mgmt import Management
+from sekoia_automation.connector import Connector
+
+from sentinelone_module.logs.configuration import SentinelOneLogsConnectorConfiguration
+
+SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE = 1
+
+# Declare prometheus metrics
+prom_namespace = "sicconfapi_intakes"
+
+OUTCOMING_EVENTS = Counter(
+    name="forwarded_events",
+    documentation="Number of events forwarded to SEKOIA.IO",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+FORWARD_EVENTS_DURATION = Histogram(
+    name="forward_events_duration",
+    documentation="Duration to collect and forward events",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+
+class SentinelOneLogsConnector(Connector):
+    configuration: SentinelOneLogsConnectorConfiguration
+
+    @cached_property
+    def management_client(self):
+        return Management(hostname=self.configuration.management_domain, api_token=self.configuration.api_key)
+
+    @property
+    def logs_retrieval_start_date(self) -> str:
+        return (
+            datetime.now(tz=timezone.utc) - timedelta(days=SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE)
+        ).isoformat()
+
+    def validate_settings(self):
+        if not self.configuration.api_key:
+            self.log(f"Connector {self.configuration.uuid} missing api_key value", level="warning")
+            return
+
+        if not self.configuration.management_domain:
+            self.log(f"Connector {self.configuration.uuid} missing management_domain value", level="warning")
+            return
+
+        try:
+            self.management_client.system.get_info()
+            self.configuration.validated = True
+            self.configuration.last_activity_created_at = self.logs_retrieval_start_date
+            self.configuration.last_threat_created_at = self.logs_retrieval_start_date
+
+        except UnauthorizedException as unauthorized_exception:
+            self.log(f"Connector {self.configuration.uuid} unauthorized to retrieve system info", level="warning")
+            raise unauthorized_exception
+
+    def pull_all_content(self):
+        start_time = time.time()
+
+        # Pull Activities
+        self.pull_activities()
+
+        # Pull Threats
+        self.pull_threats()
+
+        FORWARD_EVENTS_DURATION.labels(intake_key=self.configuration.intake_key, datasource="sentinelone").observe(
+            time.time() - start_time
+        )
+
+    def pull_activities(self):
+        query_filter = ActivitiesFilter()
+        query_filter.apply(key="limit", val=1000)
+        query_filter.apply(key="sortBy", val="createdAt")
+        query_filter.apply(key="sortOrder", val="asc")
+
+        while True:
```

Comment:
For infinity loop, use `self.running` that allow gracefully shutdown the connector
```suggestion
        while self.running:
```